shell/platform/darwin/macos/framework/Source/FlutterEngine_Internal.h

```diff
@@ -21,6 +22,11 @@
  */
 @property(nonatomic, readonly, nullable) NSOpenGLContext* resourceContext;
 
+/**
+ * The `FlutterMacOSCompositor` associated with this engine, if any.
+ */
+@property(nonatomic, nullable) flutter::FlutterMacOSCompositor* macOSCompositor;
```

Comment:
make this `std::unique_ptr<flutter::FlutterMacOSCompositor>` that way we don't need to worry about memory management.